### PR TITLE
Additional assert statement for dims alignment

### PR DESCRIPTION
**Pull request by recycletechno**
Propose adding an additional assert statement in the `MaskedSelfAttention` class to verify that the number of attention heads matches the dim size. Otherwise, if `self.dim` is not divisible by `self.num_heads` without a remainder, there may be subtle and hard-to-detect errors inside `nn.Module()` when calling the `TransformerLayer` class in the future.

### CODE REVIEW

**Comment by yhn112** [comment]
LGTM
